translib/transformer/subscribe_req_xlate.go

```diff
@@ -1052,13 +1052,26 @@ func (pathXltr *subscribePathXlator) handleYangToDbKeyXfmr() (string, error) {
 			if len(pathElems[idx].Key) > 0 {
 				if ygXpathInfo, ok := xYangSpecMap[ygPath]; ok {
 					if ygXpathInfo.virtualTbl == nil || !(*ygXpathInfo.virtualTbl) {
-						for _, kv := range pathElems[idx].Key {
+						/* Build the database key string in the form of:
+						 * <keyVal1><Seperator><KeyVal2>...<Seperator><KeyValN>
+						 * or just <keyVal1> in the case of a single key.
+						 * Note the yangEntry.Key field contains the key field names in order and
+						 * seperated by ' ' characters while the pathElems.Key variable maps the
+						 * field names to their values. */
+						for _, keyName := range strings.Split(ygXpathInfo.yangEntry.Key, " ") {
+							val, ok := pathElems[idx].Key[keyName]
+							// Use a wildcard when the key value isn't specified.
+							kv := "*"
+							if ok {
```

Comment:
Always the value of the key will be either specific or wild card, so remove this default value assignment. In case if there is  a value which is neither specific or wild card, then we need to fix the root cause of the issue, not here by assigning wild card value.

Reply:
@mbalachandar thanks for the feedback.  We've not seen any instances where this default case was needed.  I've updated the change based on your comment.